### PR TITLE
[feat]: [CDS-33860]: Error handling for K8s CLI errors

### PR DESCRIPTION
**Pull request by abhinavsingh1196**
Changes:
- Some more error handling cases added for k8s CLI errors.

You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/31128)

### CODE REVIEW

**Review by abosii-harness — CHANGES_REQUESTED** [review]
> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31128)* status: 1 change requests. 0 of 1 approvals obtained. 

___
*[930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java, line 1927 at r1](https://reviewable.io/reviews/harness/harness-core/31128#-Mx9pHWu92tFyVXv4Yik:-Mx9pHWu92tFyVXv4Yil:b-q8auai) ([raw file](https://github.com/harness/harness-core/blob/2af7a7373642ce107a1daa28a88f2a5d5c691d95/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java#L1927)):*
> ```Java
>           throw NestedExceptionUtils.hintWithExplanationException(KubernetesExceptionHints.MANIFEST_RENDER_ERROR,
>               format(KubernetesExceptionExplanation.MANIFEST_RENDER_ERROR, errorCaptureStream.toString()),
>               new InvalidRequestException(
> ```

Can we KubernetesTaskException instead of InvalidRequestException?

**Comment by abhinavsingh1196** [comment]
___
*[930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java, line 1927 at r1](https://reviewable.io/reviews/harness/harness-core/31128#-Mx9pHWu92tFyVXv4Yik:-MxDYoWL4h4CC5tJCrOy:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/2af7a7373642ce107a1daa28a88f2a5d5c691d95/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java#L1927)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

Can we KubernetesTaskException instead of InvalidRequestException? 
</blockquote></details>

Done.

**Review by abosii-harness — APPROVED** [review]
> Reviewed 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31128)* status: :shipit: complete! ✓

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java [hunk 1]
```diff
@@ -1892,7 +1892,9 @@ public void savingPatchesToDirectory(
     } catch (KubernetesValuesException kvexception) {
       String message = kvexception.getParams().get("reason").toString();
       executionLogCallback.saveExecutionLog(message, ERROR);
-      throw new KubernetesValuesException(message, kvexception.getCause());
+      throw NestedExceptionUtils.hintWithExplanationException(
+          "Please check if the given values yaml file(s) are valid YAMLs.", "Values yaml file(s) are not valid.",
```
Comment (shubham-in10se) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java:1896]: lets move these messages to the `KubernetesExceptionHints` class

**Review by shubham-in10se — APPROVED** [review]

**Review by abosii-harness — APPROVED** [review]
> Reviewed 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31128)* status: :shipit: complete! ✓

**Review by shubham-in10se — APPROVED** [review]

**Pull request merged**